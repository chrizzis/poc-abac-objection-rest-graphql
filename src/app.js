import express from 'express'
import path from 'path'
import cors from 'cors'
import attachCurrentUser from './http-middleware/attachCurrentUser.js'
import { apiUrl } from '../config/index.js'
import userRoute from './routes/user.route.js'
import projectRoute from './routes/project.route.js'
import authRoute from './routes/auth.route.js'
const __dirname = path.resolve()

const app = express()

app.use(cors())
app.use(express.urlencoded({ extended: false }))
app.use(express.json())
app.use(express.static(path.join(__dirname, `/public`)))

app.use(authRoute)
app.use(apiUrl, attachCurrentUser)
app.use(apiUrl, userRoute)
app.use(apiUrl, projectRoute)

export default app