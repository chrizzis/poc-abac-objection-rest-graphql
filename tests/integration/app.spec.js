import request from 'supertest'
import app from '../../src/app.js'

describe("app.js", () => {
  let server
  beforeAll(async () => {
  })
  describe("Index route", () => {
    test("It should return status 200", async () => {
      const response = await request(app).get(`/`);
      expect(response.statusCode).toBe(200);
    });
    test("should return the static site from the index route", async () => {
      const response = await request(app).get(`/`);

      expect(response.type).toEqual("text/html");
      expect(response.text).toMatchSnapshot()
    });

    test("should return status 404 for undefined route", async () => {
      const response = await request(app).get("/nonexistentroute").expect(404);
    });
  });
});