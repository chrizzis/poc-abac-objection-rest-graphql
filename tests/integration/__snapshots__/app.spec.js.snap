// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.js Index route should return the static site from the index route 1`] = `
"<!doctype html>
<html>

<head>
  <title>This is the title of the webpage!</title>
</head>

<body>
  <p>This is an example paragraph. Anything in the <strong>body</strong> tag will appear on the page, just like this
    <strong>p</strong> tag and its contents.
  </p>
</body>

</html>"
`;
